pkg/rules/rules.go

```diff
@@ -51,6 +51,32 @@ func (r RuleNamespace) LintPromQLExpressions() (int, int, error) {
 	return count, mod, nil
 }
 
+// LintRecordingRules checks that recording rules have at least one colon in their name, this is based
+// on the recording rules best practices here: https://prometheus.io/docs/practices/rules/
+// Returns the number of rules that don't match the requirements, and error if that number is not 0.
+func (r RuleNamespace) LintRecordingRules() int {
+	var name string
+	var count int
+	for _, group := range r.Groups {
+		for _, rule := range group.Rules {
+			name = getRuleName(rule)
```

Comment:
I believe we can't use `getRuleName` here as it will also target alerts. How do you feel about assuming that if `rule.Record` is `""` this is an alert and not a recording rule. You could verify both: the absence of `Record` and presence of `Alert` - the absence of both would be an error.